### DISCUSSION TITLE
Can you SET node values from within a plugin?

### DISCUSSION THREAD

**Discussion by Westcott1** [zwave-js/zwave-js-ui#1961] [Q&A]
I'm writing a plugin to send MQTT directly to Domoticz (domoticz/in).
If the plugin also listens for MQTT updates from Domoticz (domoticz/out/+) can these SET values?

**Comment by robertsLando** [comment]
> If the plugin also listens for MQTT updates from Domoticz (domoticz/out/+) can these SET values?

Sure

**Reply by Westcott1** [reply]
Thanks for the confirmation.
Is there any documentation on how to do this?

**Reply by robertsLando** [reply]
@Westcott1 You should init an mqtt client that listens to those topics and forwards write requests to zwave client

**Reply by Westcott1** [reply]
Thanks.

> You should init an mqtt client that listens to those topics

Yes, that is exactly what I am doing.

> forwards write requests to zwave client

Is there any documentation or code example to show how it is done  in a plugin?
Are these perhaps methods available on the plugin's ctx.zwave object?

Thanks again.

**Reply by robertsLando** [reply]
In plugin context you also have MqttClient instance: https://github.com/zwave-js/zwavejs2mqtt/blob/f5832b9a90c9629fad0b9c941d2a7817ef816a6c/lib/CustomPlugin.ts#L8

You could use it (but mind that it only emits events for topics that end with /set). Otherwise nope I don't have any example of working plugins using mqtt, there are 3/4 plugins out there for z2m AFAIK (and them are listed in docs)
